java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -3900,4 +3901,52 @@ public void bitpos(BaseClient client) {
             assertTrue(executionException.getCause() instanceof RequestException);
         }
     }
+
+    @SneakyThrows
+    @ParameterizedTest(autoCloseArguments = false)
+    @MethodSource("getClients")
+    public void lmpop(BaseClient client) {
+        assumeTrue(REDIS_VERSION.isGreaterThanOrEqualTo("7.0.0"), "This feature added in redis 7");
+        // setup
+        String key1 = "{key}-1" + UUID.randomUUID();
+        String key2 = "{key}-2" + UUID.randomUUID();
+        String nonListKey = "{key}-3" + UUID.randomUUID();
+        String[] singleKeyArray = {key1};
+        String[] multiKeyArray = {key2, key1};
+        long count = 1L;
+        Long arraySize = 5L;
+        String[] lpushArgs = {"one", "two", "three", "four", "five"};
+        Map<String, String[]> expected = Map.of(key1, new String[] {"five"});
+        Map<String, String[]> expected2 = Map.of(key2, new String[] {"one", "two"});
+
+        // nothing to be popped
+        assertNull(client.lmpop(singleKeyArray, PopDirection.LEFT).get());
+        assertNull(client.lmpop(singleKeyArray, PopDirection.LEFT, count).get());
+
+        // pushing to the arrays to be popped
+        assertEquals(arraySize, client.lpush(key1, lpushArgs).get());
+        assertEquals(arraySize, client.lpush(key2, lpushArgs).get());
+
+        // assert correct result from popping
+        Map<String, String[]> result = client.lmpop(singleKeyArray, PopDirection.LEFT).get();
+        assertEquals(Map.of(key1, new String[] {"five"}).keySet(), result.keySet());
+        for (String k : expected.keySet()) {
+            assertArrayEquals(expected.get(k), result.get(k));
+        }
```

Comment:
use deep compare instead of these lines
btw `Map.of(key1, new String[] {"five"}).keySet()` === `Set.of(key1)`